### PR TITLE
Show correct aria-expanded state for mobile-menu-button

### PR DESCRIPTION
**Pull request by Rleahy22**
## Description
See: https://github.com/department-of-veterans-affairs/va.gov-team/issues/25659#issuecomment-917015823

Previously the aria-expanded attribute was always false. Additionally there were two separate buttons that were shown/hidden based on the display state, but I collapsed them to one, and in doing so no longer needed a class component, and switched to a pure function.

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/25659

## Testing done
local, unit

## Screenshots

https://user-images.githubusercontent.com/3144003/133317437-e0f82dd2-69aa-4b64-9c78-5403be840382.mov

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by Rleahy22** [comment]
@joshkimux you can see in the screen recording that it now updates, but if you see the expanded/collapsed text disappears really quickly so you just see/hear "Open" or "closed". Is that okay? I don't know why it's doing that.

**Comment by joshkimux** [comment]
@Rleahy22 Couple of things:

- When using VO, try and use Safari! Chrome is known to be a bit buggy, that might explain the issue you're having
- `aria-expanded` will communicate the state of being open or closed, so I'd recommend keeping the label as "menu" instead of "open"

**Comment by Rleahy22** [comment]
> @Rleahy22 Couple of things:
> 
> * When using VO, try and use Safari! Chrome is known to be a bit buggy, that might explain the issue you're having
> * `aria-expanded` will communicate the state of being open or closed, so I'd recommend keeping the label as "menu" instead of "open"

Annnnnd it works!

https://user-images.githubusercontent.com/3144003/133660133-b082e8fd-655b-4e90-beab-babd4085eacf.mov

**Comment by joshkimux** [comment]
@Rleahy22 Brilliant! This will be huge for accessibility 👏

File: src/platform/site-wide/mobile-menu-button/tests/main.unit.spec.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      userEvent.click(toggleButton);
+      expect(toggleButton.getAttribute('aria-expanded')).to.eq('true');
+      expect(screen.queryByText('Open')).to.eq(null);
+
+      userEvent.click(toggleButton);
+      expect(toggleButton.getAttribute('aria-expanded')).to.eq('false');
+      expect(screen.queryByText('Close')).to.eq(null);
+    });
+  });
+});
```
Comment (k80bowman) [inline src/platform/site-wide/mobile-menu-button/tests/main.unit.spec.js:29]: Thank you for adding tests! 🏆

**Review by k80bowman — APPROVED** [review]

**Pull request merged**